In [58]:
import markovify
from nltk.sentiment import SentimentIntensityAnalyzer

with open("data/shakespear.txt") as file:
    text = file.read()

model = markovify.Text(text)

text = []

sentimentAnalyzer = SentimentIntensityAnalyzer()
for _ in range(14):
    sentence = model.make_short_sentence(70)

    while sentence == None:
        sentence = model.make_short_sentence(70)

    text.append([sentence.split(" ")])
    
    scores = sentimentAnalyzer.polarity_scores(sentence)
    scoreString = ""
    for i in sorted(scores):
       scoreString += '{0}: {1}, '.format(i, scores[i])

    print("{0:<80} - {1:<80}".format(sentence, scoreString))

poetry = "\n".join([" ".join(sent[0]) for sent in text])

scores = sentimentAnalyzer.polarity_scores(poetry)
scoreString = ""
for i in sorted(scores):
    scoreString += '{0}: {1}, '.format(i, scores[i])

print(scoreString)

But wherefore do not so; but since I am old?                                     - compound: 0.0, neg: 0.0, neu: 1.0, pos: 0.0,                                    
Your monument shall be the same.                                                 - compound: 0.0, neg: 0.0, neu: 1.0, pos: 0.0,                                    
Let me not with thine eyes.                                                      - compound: 0.0, neg: 0.0, neu: 1.0, pos: 0.0,                                    
Take heed, dear heart, of this excess, Eat up thy charge?                        - compound: 0.3818, neg: 0.0, neu: 0.794, pos: 0.206,                             
O! change thy thought, that I before have heard them told.                       - compound: 0.0, neg: 0.0, neu: 1.0, pos: 0.0,                                    
That time of year thou mayst prove me.                                           - compound: 0.0, neg: 0.0, neu: 1.0, pos: 0.0,                                    
My reason, the p

In [59]:
import gensim.downloader 

gensim = gensim.downloader.load("word2vec-google-news-300")

In [60]:
def most_similar_word(word):
    try:
        similar_words = gensim.most_similar(word)
        return similar_words[0][0]
    except KeyError:
        return word


In [61]:
from random import randint

modified = []

for sente in text:
    sente = sente[0]
    index = randint(0, len(sente) - 1)
    word = most_similar_word(sente[index])
    
    tempSent = []
    result = ""
    for i in range(len(sente)):
        
        if i == index:
            if word == sente[i]:
                result += "\033[91m{} \033[00m".format(word)
            else: 
                result += "\033[96m{} \033[00m".format(word)
            tempSent.append(word)
        else: 
            result += "{} ".format(sente[i])
            tempSent.append(sente[i])

    
    modified.append(tempSent)
    print(result)

But wherefore do not so; but since I 'm old? 
Your statue shall be the same. 
Let me not with thy eyes. 
Take heed, dear heart, of this excess, Eat up thy charge? 
O! change thine thought, that I before have heard them told. 
That time of year thou mayst prove me. 
My reason, the physician to my purposes bred, That all the Muses filed. 
Thus policy in love, to thee resort. 
Against my loved alone. 
Steve how with my desire keep pace. 
If my dear love myself rise and fall. 
How several a vanished sight: Then can no horse with my desire keep pace. 
O benefit of rest? 
But do thine worst all best exceeds? 


In [62]:
from nltk.translate.bleu_score import corpus_bleu

print("BLEU score = " + str(corpus_bleu(text, modified)))


BLEU score = 0.7805014090508074


Best one yet:

Now all is done, have what shall be the praise.                                  - compound: 0.5574, neg: 0.0, neu: 0.714, pos: 0.286,                             
Then, if for my sin you did impute, Which shall be the same.                     - compound: -0.5574, neg: 0.231, neu: 0.769, pos: 0.0,                            
Oh sure I am shamed by that which still doth grow?                               - compound: -0.3182, neg: 0.259, neu: 0.576, pos: 0.165,                          
No love toward others in that I cannot know thy change.                          - compound: 0.4588, neg: 0.153, neu: 0.556, pos: 0.292,                           
The statute of thy husbandry?                                                    - compound: 0.0, neg: 0.0, neu: 1.0, pos: 0.0,                                    
Accuse me thus: that I am gone.                                                  - compound: -0.2023, neg: 0.265, neu: 0.735, pos: 0.0,                            
So is my love, not for myself, no quiet find.                                    - compound: 0.747, neg: 0.0, neu: 0.557, pos: 0.443,                              
Proud of this madding fever!                                                     - compound: 0.5255, neg: 0.0, neu: 0.541, pos: 0.459,                             
Take heed, dear heart, of this excess, Eat up thy charge?                        - compound: 0.3818, neg: 0.0, neu: 0.794, pos: 0.206,                             
Unthrifty loveliness, why dost thou too, and therein dignified.                  - compound: 0.4939, neg: 0.0, neu: 0.714, pos: 0.286,                             
Oh sure I am gone.                                                               - compound: 0.3182, neg: 0.0, neu: 0.566, pos: 0.434,                             
Herein lives wisdom, beauty, and thy love's use their treasure.                  - compound: 0.8555, neg: 0.0, neu: 0.427, pos: 0.573,                             
Him have I slept in your sweet form should bear.                                 - compound: 0.4588, neg: 0.0, neu: 0.727, pos: 0.273,                             
Have I not to give them from me now.                                             - compound: 0.0, neg: 0.0, neu: 1.0, pos: 0.0,                  